lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -755,6 +755,12 @@ defmodule ExUnit.Assertions do
         1 + "test"
       end
 
+  To test a function that requires arguments, wrap it:
+
+      assert_raise ArithmeticError, fn ->
+        my_add(1, "test")
+      end
+
```

Comment:
WDYT about

```suggestion
  Or if the anonymous function explicitly raises:

      assert_raise RuntimeError, fn ->
        raise "this assertion will pass due to this raise"
      end

```


?

Reply:
It's a good example of raising an exception, but doesn't answer the original question I had, _"How do I test my function that requires arguments?"_

Reply:
Right. My concern is that there is nothing special to `assert_raise` as an answer to that question. You can invoke any code inside `fn -> ... end`, regardless if using `assert_raise` or not. So I was trying to find a way to add new information while answering your question, but I guess it was not enough. :)

Reply:
Yes, I see what you mean.
Let me think on it.

Reply:
Would this be too clunky?

```suggestion
  Calling the `raise/1` function:

      assert_raise RuntimeError, fn ->
        raise "assertion will pass due to this raise"
      end

```